api/services/DMStore.ts

```diff
@@ -111,36 +120,30 @@ export async function getDocumentVersionThumbnail(documentId: string, versionId:
     return response.data
 }
 
-/**
- * DOCUMENT CREATION
- */
-
-// Creates a list of Stored Documents by uploading a list of binary/text files.
-export async function postDocument(file, classification) {
-    const body: any = {}
-    body.formData = {
-        classification: getClassification(classification),
-        files: [
-            {
-                options: { filename: file.name, contentType: file.type },
-                value: fs.createReadStream(file.path),
+export function postUploadedDocument(file, classification, options) {
+    console.log('postUploadedDocument');
+    const reqOptions = {
+        ...options, ...{
+            formData: {
+                classification: getClassification(classification),
+                files: [
+                    {
+                        options: {filename: file.name, contentType: file.type},
+                        value: fs.createReadStream(file.path),
+                    },
+                ],
             },
-        ],
-    }
-
-    const response = await asyncReturnOrError(
-        http.post(`${url}/documents`, body, {
             headers: {
-                contentType: file.type,
+                ...options.headers, ...{
+                    'Content-Type': 'multipart/form-data',
+                },
             },
-        }),
-        `Error posting document`,
-        null,
-        logger,
-        false
-    )
+        },
+        json: false,
+    }
 
-    return response.data
+    //TODO: Once working replace with asyncReturnOrError
+    return generateRequest('POST', `${url}/documents`, reqOptions)
```

Comment:
I initially used the old generateRequest to get this working, but if we can take a look at using axios instead that would be good (as we want to replace all request-promise)

Reply:
+1 yes

Reply:
Axios was hard to get up and running with file upload + form data in a timely manner. Using promise-request in a seperate file - uploadFileFormData, in help to deprecate 
generateRequest()